torch/fx/experimental/symbolic_shapes.py

```diff
@@ -348,6 +336,24 @@ def constrain_range(a, *, min: Optional[int], max: Optional[int] = None):
         builtins.max(r.lower, min), builtins.min(r.upper, max)
     )
 
+def constrain_range_non_symint(a, *, min, max):
```

Comment:
This is a strange function, feels like it should be private.

Also, the name is misleading, it's not really constraining anything, but rather creating a symint.

Should you be returning that symint? In that case, you can rename the function to be something like `create_*` or `convert_*`.

Reply:
Context for that code is in https://github.com/pytorch/pytorch/pull/101655, but I'll leave it up to Xuan on how he wants to separate this out into a function.

Reply:
How about we rename it to something like `constrain_range_eager`?

Reply:
I still don't like the name. Something like `check_range_and_create_symbol` would be more appropriate.

Reply:
Agreed with @avikchaudhuri . I think `constrain_range_eager` is little vague. `check_range_and_create_symbol` sounds better.